cli/src/pcluster/config/imagebuilder_config.py

```diff
@@ -0,0 +1,149 @@
+        parameter_store_name_output: str = None,
+        root_volume: ImageBuilderVolumeConfig = None,
+        components: ImageBuilderComponentsConfig = None,
+    ):
+        super().__init__()
+        self.name = name
+        self.description = description
+        self.tags = tags
+        self.parameter_store_name_output = parameter_store_name_output
+        self.root_volume = root_volume
+        self.components = components
+        # TODO: add validator
+        self._validators = []
+
+
+class ImageBuilderBuildConfig(Config):
+    """Represent the build configuration for the ImageBuilder."""
+
+    def __init__(
+        self,
+        instance_type: str,
+        parent_image: str,
+        instance_role: str = None,  # TODO: auto generate if not assigned
+        subnet_id: str = None,  # TODO: auto generate if not assigned
+        terminate_on_failure: bool = True,
+        bucket_logs: ImageBuilderBucketLogConfig = None,
+        tags: Dict[str, str] = None,
+        security_groups: List[str] = None,
+    ):
+        super().__init__()
+        self.instance_type = instance_type
+        self.parent_image = parent_image
+        self.instance_role = instance_role
+        self.bucket_logs = bucket_logs
+        self.terminate_on_failure = terminate_on_failure
+        self.tags = tags
+        self.subnet_id = subnet_id
+        self.security_groups = security_groups
+        # TODO: add validator
+        self._validators = []
+
+
+class ImageBuilderDevSettingsConfig(Config):
+    """Represent the dev settings configuration for the ImageBuilder."""
+
+    def __init__(
+        self,
+        update_and_reboot: bool = False,
+        pcluster_component_data: str = None,
+        chef_cookbook: ImageBuilderChefCookbookConfig = None,
+        node_url: str = None,
+        aws_batch_cli_url: str = None,
+        image_recipe_arn: str = None,
+        distribution_configuration_arn: str = None,
+    ):
+        super().__init__()
+        self.update_and_reboot = update_and_reboot
+        self.pcluster_component_data = pcluster_component_data
+        self.chef_cookbook = chef_cookbook
+        self.node_url = node_url
+        self.aws_batch_cli_url = aws_batch_cli_url
+        self.image_recipe_arn = image_recipe_arn
+        self.distribution_configuration_arn = distribution_configuration_arn
+        # TODO: add validator
+        self._validators = []
+
+
+class ImageBuilderConfig(Config):
+    """Represent the configuration of an ImageBuilder."""
+
+    def __init__(
+        self,
+        image_config: ImageBuilderImageConfig,
+        build_config: ImageBuilderBuildConfig,
+        dev_settings_config: ImageBuilderDevSettingsConfig = None,
+    ):
+        super().__init__()
+        self.image_config = image_config
+        self.build_config = build_config
+        self.dev_settings_config = dev_settings_config
+        self._register_validator(BaseAMIValidator, priority=100, ami_id=self.build_config.parent_image)
```

Comment:
If you don't need a specific order for the validators you can avoid to set the priority.